### PR TITLE
Update IRIDA documentation publishing strategy

### LINKED ISSUE #798: Migrate documentation to Github pages or similar
## What needs changed?
Migrate the documentation pages to use something like GitHub pages, readthedocs, or similar platform.

## How does it work now?
Currently our documentation is generated with `mvn site` and hosted as static pages.  This generates both javadoc and markdown docs with Jekyll.

## How should it work?
Host the documentation on something like GitHub pages.  This may need to be in a separate repository because the javadoc needs to be built before it's able to be hosted.  GitHub pages uses Jekyll to generate its pages from markdown so this should be a relatively smooth transition.

## Additional information
Additional files are hosted on our webserver which will need to be relocated as well.  Some additional downloads, vm files, etc.

Part of this work will involve updating links across our other files and sites to point to this new docs location, particularly irida.ca, our changelog/upgrading files, some places in the IRIDA codebase, notifications in Gitter, etc.

### PR DESCRIPTION
**Pull request by tom114**
## Description of changes
Created a new documentation publishing plan for IRIDA's documentation.  The new docs are published using a repository at <https://github.com/phac-nml/irida-documentation/> and GitHub pages.  The current docs have been updated and are available at <https://phac-nml.github.io/irida-documentation/>.

This is a replacement for the PR at #891.  That PR completely removed the docs from this repository.  We decided to keep them in this repo for development purposes but publish out to the external repo.  This way we can ensure that code changes and documentation changes are updated at the same time.

To test this PR I suggest checking out the new docs pages repo at https://github.com/phac-nml/irida-documentation/ and going through the update procedure in the documentation pages (not pushing all the way up, but the export, etc).

## Related issue
Fixes #798

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: build-docs.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    print("Cannot find pom file.  This script must be run in the base directory of the IRIDA repository.")
+    exit(1)
+
+if not os.path.isdir(doc_dir):
+    print("Cannot find `/doc` directory.  This script must be run in the base directory of the IRIDA repository.")
+    exit(1)
+
+# check if the argument is the irida-documentation/docs directory
+if not (result_dir.endswith('docs') or result_dir.endswith('docs/')) or not os.path.isdir(result_dir):
+    print("The target directory doesn't appear to be the irida-docuemtation repository's `/docs` directory.")
```
Comment (deepsidhu85) [inline build-docs.py:34]: Typo in `documentation`

File: build-docs.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+if retval != 0:
+    print("========== Docs construction failed.  See above for error messages")
+    exit(1)
+
+
+print("========== Documentation built successfully")
+
+# Give the user 5 seconds to chicken out
+print("========== WARNING: this script will now replace the existing docs directory {} with the docs built above".format(result_dir))
+print("========== This process will contine in 5 seconds...")
```
Comment (deepsidhu85) [inline build-docs.py:51]: Typo in continue

Comment (deepsidhu85) [inline build-docs.py:51]: Provide the user with the command to exit out if they do not want to continue

File: doc/developer/getting-started/index.md [hunk 3]
```diff
@@ -263,24 +263,44 @@ mvn clean package -DskipTests
...
 
 #### Building IRIDA documentation
+IRIDA documentation can be found in the <https://github.com/phac-nml/irida-docs> GitHub project.  IRIDA's documentation is built using [Jekyll][] and [GitHub Pages](https://pages.github.com/).  Jekyll allows us to write documentation in Markdown format and it will convert the pages to HTML.  We can use Jekyll both for viewing the documentation locally and for publishing to GitHub Pages.  The current documentation can be found at <https://phac-nml.github.io/irida-docs>.
 
-IRIDA documentation can be found in the `doc/` directory in the IRIDA root directory.  IRIDA's documentation is built using [Jekyll][].  Jekyll allows us to write documentation in Markdown format and it will convert the pages to HTML for releasing to the web.  The documentation at [http://irida.corefacility.ca/documentation](http://irida.corefacility.ca/documentation) is all generated using this tool.
+##### Testing IRIDA documentation locally
+To view the documentation locally or make changes, you can checkou the above GitHub project and make changes.  To run the server locally you can run Jekyll to generate the pages.
```
Comment (deepsidhu85) [inline doc/developer/getting-started/index.md:269]: Typo in checkout

File: doc/developer/getting-started/index.md [hunk 4]
```diff
@@ -263,24 +263,44 @@ mvn clean package -DskipTests
...
 
-To test any documentation changes, you can `cd` into the `doc/` directory and run the following command:
+First you can `cd` into the `docs/` directory and run the following command:
```
Comment (deepsidhu85) [inline doc/developer/getting-started/index.md:271]: Change `First you can` to just `First`

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Still need to try it out but I left some comments below

File: doc/developer/getting-started/index.md (hunk 4 above)
Comment (tom114) [reply doc/developer/getting-started/index.md]: Fixed in 6351ffb

File: doc/developer/getting-started/index.md (hunk 3 above)
Comment (tom114) [reply doc/developer/getting-started/index.md]: Fixed in 6351ffb

File: build-docs.py (hunk 2 above)
Comment (tom114) [reply build-docs.py]: Fixed in 6351ffb

Comment (tom114) [reply build-docs.py]: Fixed in 6351ffb

File: build-docs.py (hunk 1 above)
Comment (tom114) [reply build-docs.py]: Fixed in 6351ffb

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**